### PR TITLE
Don't log warning when failing to encode non-default bucketspace

### PR DESCRIPTION
**Pull request by toregge**
for storage::api::RequestBucketInfoCommand.

@geirst, @vekterli: please review

### CODE REVIEW

File: storageapi/src/vespa/storageapi/mbusprot/storageprotocol.cpp [hunk 1]
```diff
@@ -110,10 +119,13 @@ StorageProtocol::encode(const vespalib::Version& version,
         }
 
     } catch (std::exception & e) {
-        LOGBP(warning, "Failed to encode %s storage protocol message %s: %s",
-              version.toString().c_str(),
-              message.getInternalMessage()->toString().c_str(),
-              e.what());
+        if (version < version6_0 &&
+            !suppressEncodeWarning(message.getInternalMessage().get())) {
```
Comment (vekterli) [inline storageapi/src/vespa/storageapi/mbusprot/storageprotocol.cpp]: With this we cannot get any warnings if other encodes fail on >= `version6_0`. Do we instead want
```c++
if (!(version < version6_0 && suppressEncodeWarning(message.getInternalMessage().get()))) {
```
?

**Review by vekterli — COMMENTED** [review]
See question on logging condition

**Review by vekterli — APPROVED** [review]
:+1:

**Review by geirst — COMMENTED** [review]
👍

**Pull request merged**